src/sql/magic.py

```diff
@@ -619,6 +622,73 @@ def _persist_dataframe(
         display.message_success(f"Success! Persisted {table_name} to the database.")
 
 
+def find_toml_path():
+    """
+    Recursively finds an absolute path to pyproject.toml starting
+    from current directory
+    """
+    current = Path().resolve()
+    while not (current / "pyproject.toml").exists():
+        if current == current.parent:
+            return None
+
+        current = current.parent
+    display.message(f"Found pyproject.toml from '{current}'")
+
+    return str(current) + "/pyproject.toml"
+
+
+def get_configs(file_path):
+    """Returns saved configs from given file_path"""
+    with open(file_path, "r") as file:
+        content = file.read()
+        data = toml.loads(content)
+    try:
+        return data["tool"]["jupysql"]["SqlMagic"]
+    except KeyError:
+        return {}
+
+
+def get_config_variables(obj):
+    """Returns a list of configs users can set"""
+    config_vars = list(vars(obj)["_all_trait_default_generators"].keys())
+    config_vars.remove("config")
+    config_vars.remove("parent")
+    return config_vars
+
+
+def load_configs(ip):
+    """Loads saved configs in pyproject.toml"""
+    file_path = find_toml_path()
+    if file_path:
+        configs = get_configs(file_path)
+        config_vars = get_config_variables(SqlMagic)
+        for config, value in configs.items():
+            if config in config_vars:
```

Comment:
This can be enclosed in a function like `config_found`

Reply:
It is now enclosed in `set_configs` in `magic.py`.